tests/catch/unit/device/hipDeviceSetGetMemPool.cc

```diff
@@ -0,0 +1,165 @@
+  if (!mem_pool_support) {
+    HipTest::HIP_SKIP_TEST("Test only runs on devices with memory pool support");
+    return false;
+  }
+  return true;
+}
+
+static inline hipMemPool_t CreateMemPool(const int device) {
+  hipMemPoolProps kPoolProps;
+  kPoolProps.allocType = hipMemAllocationTypePinned;
+  kPoolProps.handleTypes = hipMemHandleTypeNone;
+  kPoolProps.location.type = hipMemLocationTypeDevice;
+  kPoolProps.location.id = device;
+  kPoolProps.win32SecurityAttributes = nullptr;
+  memset(kPoolProps.reserved, 0, sizeof(kPoolProps.reserved));
+
+  hipMemPool_t mem_pool;
+  HIP_CHECK(hipMemPoolCreate(&mem_pool, &kPoolProps));
+
+  return mem_pool;
+}
+
+TEST_CASE("Unit_hipDeviceSetMemPool_Positive_Basic") {
+  const int device = GENERATE(range(0, HipTest::getDeviceCount()));
+
+  if (!CheckMemPoolSupport(device)) {
+    return;
+  }
+
+  hipMemPool_t mem_pool = CreateMemPool(device);
+  HIP_CHECK(hipDeviceSetMemPool(device, mem_pool));
+
+  HIP_CHECK(hipMemPoolDestroy(mem_pool));
+}
+
+TEST_CASE("Unit_hipDeviceSetMemPool_Negative_Parameters") {
+  hipMemPool_t mem_pool;
+  HIP_CHECK(hipDeviceGetDefaultMemPool(&mem_pool, 0));
+
+  SECTION("mem_pool == nullptr") {
+    HIP_CHECK_ERROR(hipDeviceSetMemPool(0, nullptr), hipErrorInvalidValue);
+  }
+
+  SECTION("device < 0") {
+    HIP_CHECK_ERROR(hipDeviceSetMemPool(-1, mem_pool), hipErrorInvalidValue);
+  }
+
+  SECTION("device ordinance too large") {
+    HIP_CHECK_ERROR(hipDeviceSetMemPool(HipTest::getDeviceCount(), mem_pool), hipErrorInvalidValue);
+  }
+}
+
+TEST_CASE("Unit_hipDeviceGetMemPool_Positive_Default") {
+  const int device = GENERATE(range(0, HipTest::getDeviceCount()));
+
+  if (!CheckMemPoolSupport(device)) {
+    return;
+  }
+
+  hipMemPool_t default_mem_pool;
+  HIP_CHECK(hipDeviceGetDefaultMemPool(&default_mem_pool, device));
+
+  hipMemPool_t mem_pool;
+  HIP_CHECK(hipDeviceGetMemPool(&mem_pool, device));
+
+  REQUIRE(mem_pool == default_mem_pool);
+}
+
+TEST_CASE("Unit_hipDeviceGetMemPool_Positive_Basic") {
+  const int device = GENERATE(range(0, HipTest::getDeviceCount()));
+
+  if (!CheckMemPoolSupport(device)) {
+    return;
+  }
+
+  hipMemPool_t mem_pool = CreateMemPool(device);
+  HIP_CHECK(hipDeviceSetMemPool(device, mem_pool));
+
+  hipMemPool_t returned_mem_pool;
+  HIP_CHECK(hipDeviceGetMemPool(&returned_mem_pool, device));
+
+  REQUIRE(returned_mem_pool == mem_pool);
+
+  HIP_CHECK(hipMemPoolDestroy(mem_pool));
+}
+
+TEST_CASE("Unit_hipDeviceGetMemPool_Positive_Threaded") {
+  class HipDeviceGetMemPoolTest : public ThreadedZigZagTest<HipDeviceGetMemPoolTest> {
+   public:
+    void TestPart2() {
+      mem_pool_ = CreateMemPool(0);
+      HIP_CHECK_THREAD(hipDeviceSetMemPool(0, mem_pool_));
+    }
+
+    void TestPart3() {
+      hipMemPool_t returned_mem_pool;
+      HIP_CHECK(hipDeviceGetMemPool(&returned_mem_pool, 0));
+
+      REQUIRE(returned_mem_pool == mem_pool_);
+    }
+
+    void TestPart4() { HIP_CHECK_THREAD(hipMemPoolDestroy(mem_pool_)); }
```

Comment:
Remove the empty lines between TestPart2/3/4

Reply:
The suggested change is implemented. Please proceed with the review.